Podcast/EmptyStateTableView.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  EmptyStateTableView.swift
+//  Podcast
+//
+//  Created by Natasha Armbrust on 10/25/17.
+//  Copyright © 2017 Cornell App Development. All rights reserved.
+//
+
+import UIKit
+
+protocol EmptyStateTableViewDelegate: class {
+    func didPressEmptyStateViewActionItem()
+}
+
+/**
+ UITableView Subclass with predefined styles and an empty state view shown when no cells are in this tableView for all sections
+ */
+class EmptyStateTableView: UITableView, EmptyStateViewDelegate {
+    
+    var type: EmptyStateType
+    var emptyStateView: EmptyStateView!
+    weak var emptyStateTableViewDelegate: EmptyStateTableViewDelegate?
+    
+    init(withType type: EmptyStateType) {
+        self.type = type
+        super.init(frame: .zero, style: .plain)
+        emptyStateView = EmptyStateView(type: type)
+        emptyStateView.delegate = self
+        backgroundView = emptyStateView
+        backgroundView?.isHidden = false
+        showsVerticalScrollIndicator = false
+        separatorStyle = .none
+        backgroundColor = .clear
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        // make sure there is no data in tableView before displaying background 
+        for s in 0..<numberOfSections {
```

Comment:
Will this dynamically show the view if I say delete all the items? i.e. the tableView loads with elements but then is brought down to 0 elements.

Reply:
Yup.